x-pack/plugins/alerting/server/saved_objects/migrations/8.8/index.ts

```diff
@@ -73,9 +73,30 @@ function addSecuritySolutionActionsFrequency(
   return doc;
 }
 
+function unmuteSecuritySolutionCustomRules(
+  doc: SavedObjectUnsanitizedDoc<RawRule>
+): SavedObjectUnsanitizedDoc<RawRule> {
+  if (!isDetectionEngineAADRuleType(doc) || doc.attributes.params.immutable) {
```

Comment:
Why do we have a check for `doc.attributes.params.immutable`?
Can we add a comment to the code explaining this?

Reply:
> The migration is applied only to custom rules

@maximpn and what is the reason for this?

Reply:
I've eventually removed `doc.attributes.params.immutable` for simplicity so the migration will be applied to all security rules. It should be much easier to read such a migration.

Reply:
Agree 👍